client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-expanded-content/boost-site-performance.tsx

```diff
@@ -45,123 +48,160 @@ export default function BoostSitePerformance( { site, trackEvent, hasError }: Pr
-						{ hasPendingScore ? (
-							<InProgressIcon />
-						) : (
-							<div className="site-expanded-content__device-score-container">
-								<div className="site-expanded-content__card-content-column">
-									<Icon
-										size={ 24 }
-										className="site-expanded-content__device-icon"
-										icon={ jetpackBoostDesktopIcon }
-									/>
-									<span className="site-expanded-content__device-score">{ desktopScore }</span>
-								</div>
-								<div className="site-expanded-content__card-content-column site-expanded-content__card-content-column-mobile">
-									<Icon
-										className="site-expanded-content__device-icon"
-										size={ 24 }
-										icon={ jetpackBoostMobileIcon }
-									/>
-									<span className="site-expanded-content__device-score">{ mobileScore }</span>
+						<div className="site-expanded-content__card-content-column">
+							{ hasPendingScore ? (
+								<InProgressIcon />
+							) : (
+								<div className="site-expanded-content__device-score-container">
+									<div className="site-expanded-content__card-content-column">
+										<Icon
+											size={ 24 }
+											className="site-expanded-content__device-icon"
+											icon={ jetpackBoostDesktopIcon }
+										/>
+										<span className="site-expanded-content__device-score">{ desktopScore }</span>
+									</div>
+									<div className="site-expanded-content__card-content-column site-expanded-content__card-content-column-mobile">
+										<Icon
+											className="site-expanded-content__device-icon"
+											size={ 24 }
+											icon={ jetpackBoostMobileIcon }
+										/>
+										<span className="site-expanded-content__device-score">{ mobileScore }</span>
+									</div>
 								</div>
+							) }
+							<div className="site-expanded-content__card-content-score-title">
+								{ translate( 'Devices' ) }
 							</div>
-						) }
-						<div className="site-expanded-content__card-content-score-title">
-							{ translate( 'Devices' ) }
 						</div>
 					</div>
+					<div className="site-expanded-content__card-footer">
+						{ ctaButtons.map( ( ctaButton ) => (
+							<Button
+								key={ ctaButton.label }
+								href={ ctaButton.href }
+								target="_blank"
+								onClick={ ctaButton.onClick }
+								className="site-expanded-content__card-button"
+								primary={ ctaButton.primary }
+								compact
+							>
+								{ ctaButton.label } { !! ctaButton.href && <Gridicon icon="external" /> }
+							</Button>
+						) ) }
+					</div>
 				</div>
-				<div className="site-expanded-content__card-footer">
-					<Button
-						href={ buttonProps.href }
-						target="_blank"
-						onClick={ buttonProps.onClick }
-						className="site-expanded-content__card-button"
-						compact
-					>
-						{ buttonProps.label }
-					</Button>
-				</div>
-			</div>
-			{ showBoostModal && (
+			</ExpandedCard>
+
+			{ boostModalState.show && (
```

Comment:
We are moving the modal component outside the `ExpandedCard` to fix a bug with an empty card not being able to show the Boost modal due to the element not existing,